controller.go

```diff
@@ -362,6 +378,49 @@ func (c *controller) NetworkByID(id string) (Network, error) {
 	return nil, ErrNoSuchNetwork(id)
 }
 
+func (c *controller) NetContainers() []NetContainer {
+	c.Lock()
+	defer c.Unlock()
+
+	list := make([]NetContainer, 0, len(c.containers))
+	for _, s := range c.containers {
+		list = append(list, s)
+	}
+
+	return list
+}
+
+func (c *controller) WalkNetContainers(walker NetContainerWalker) {
+	for _, nc := range c.NetContainers() {
+		if walker(nc) {
+			return
+		}
+	}
+}
+
+func (c *controller) NetContainerByID(id string) (NetContainer, error) {
+	if id == "" {
```

Comment:
At the time I wrote this, I did not have `containers` element in endpoint structure, only had `sandboxes`.
Now that sandboxes are decoupled from net containers (did this so that multiple containers can use the default sandbox), we can retrieve the NetContainer as a direct key lookup in the `containersTable` map.
Will change when taking care of other comments.